wordpress-mobile/WordPress-Android
# Issue/13066 ppn category select

Part of #13066 

This PR is a refactor of `PrepublishingCategoriesViewModel`, PrepublishingCategoriesAdapter  and to a lesser extent `PrepublishingCategoriesFragment`. It addresses a bunch of architectural recommendations in PR #13150. 

in `PrepublishingCategoriesViewModel` `PrepublishingCategoriesListItemUiState` was added to maintain consistency.  The SwipeRefreshLayout was removed and that cleared up the scrolling issue. In `PrepublishingCategoriesAdapter` the ViewHolder was extracted out to a separate class file and binding is done with a uiState object. 

**Merge Instructions**
Note: This PR will be merged into a feature branch
Make sure #13150 has been merged
Make sure #13167 has been merged
Remove the Not Ready For Merge label
Merge as normal

**To test**
- Launch the app
- Select Blog posts
- Select Edit on a post
- From the edit post view, tap the UPDATE action to launch the bottom sheet
- Categories are shown on the Categories row
(If the post has categories assigned, you will see them listed on the home view. If not, you will see Not Set)
- Tap the categories row to launch the Select categories view
- Tap the <- Back arrow to save your options
- The categories row on the home screen should reflect the selections you made

PR submission checklist:
